crates/tauri-bundler/src/bundle/macos/app.rs

```diff
@@ -214,7 +214,11 @@ fn create_info_plist(
     settings.bundle_identifier().into(),
   );
   plist.insert("CFBundleInfoDictionaryVersion".into(), "6.0".into());
-  plist.insert("CFBundleName".into(), settings.product_name().into());
+  if let Some(bundle_name) = settings.macos().bundle_name.clone() {
+    plist.insert("CFBundleName".into(), bundle_name.into());
+  } else {
+    plist.insert("CFBundleName".into(), settings.product_name().into());
+  }
```

Comment:
This code can be "simplified" (not really simpler) to how it was done for CFBundleVersion below so like this
```suggestion
  plist.insert(
    "CFBundleName".into(),
    settings
      .macos()
      .bundle_name
      .as_deref()
      .unwrap_or_else(|| settings.product_name())
      .into(),
  );
```

Reply:
By the tauri test config file
https://github.com/tauri-apps/tauri/blob/6a39f49991e613e8f3befe0e8dff288482ccdd89/crates/tauri/test/fixture/src-tauri/tauri.conf.json#L1-L22
This could be passed all of the testing due to the `product_name` is `Option`


`macos().bundle_name` and `settings.product_name()`, they are both `Option`. Accesing like this gets fail test due to the tauri test config file dose not contain any `product name` or `bundle name`
```rust
plist.insert(
    "CFBundleName".into(),
    settings
      .macos()
      .bundle_name
      .as_deref()
      .unwrap_or_else(|| settings.product_name())
      .into(),
  );
```
since when `bundle_name` is `None` will fallback to `product_name`. additional `product_name` unwrap_or opperation will makes code heavy.

It could be solved to add test file config or using `if Some(...) = ... else if Some(...) ...`